src/Algebra/Free/Magma.agda

```diff
@@ -0,0 +1,441 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of 'pre-free' and 'free' magma
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Free.Magma where
+
+open import Algebra.Core
+open import Algebra.Bundles using (Magma)
+open import Algebra.Bundles.Raw using (RawMagma)
+import Algebra.Definitions as Definitions using (Congruent₂)
+import Algebra.Structures as Structures using (IsMagma)
+open import Algebra.Morphism.Bundles using (MagmaHomomorphism)
+open import Algebra.Morphism.Structures using (IsMagmaHomomorphism)
+import Algebra.Morphism.Construct.Identity as Identity
+import Algebra.Morphism.Construct.Composition as Compose
+open import Effect.Functor
+open import Effect.Monad
+open import Function.Base using (id; _∘_)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Morphism using (IsRelHomomorphism)
+open import Level using (Level; suc; _⊔_)
+open import Relation.Nullary.Negation.Core using (¬_)
+open import Relation.Binary
+  using (Setoid; IsEquivalence; Reflexive; Symmetric; Transitive)
+open import Relation.Binary.Morphism.Bundles using (SetoidHomomorphism)
+import Relation.Binary.Morphism.Construct.Identity as Identity
+import Relation.Binary.Morphism.Construct.Composition as Compose
+open import Relation.Binary.PropositionalEquality
+  using (_≡_; cong₂; _≗_)
+  renaming (refl to ≡-refl; isEquivalence to ≡-isEquivalence)
+import Relation.Binary.Reasoning.Setoid as ≈-Reasoning
+
+private
+  variable
+    ℓ a ℓa b ℓb c ℓc m ℓm n ℓn : Level
+    A : Set a
```

Comment:
Note that variables are scoped / generative (or whatever the right name is). What this means is that you can do
```agda
     A B C : Set ℓ
 ```
 and at use site, each will have different `Level` automatically. The same idea can help you shorten the `Setoid` and `Magma` declarations to be 1-liners too.

Reply:
Interesting! Thanks for the nudge. 
I'll (try to remember to) implement the idea in subsequent versions/variations, but not now, I think.